code/debug/debug.cpp

```diff
@@ -0,0 +1,39 @@
+#include <algorithm>
+#include <iostream>
+
+void swap(int *a, int*b) {
+    int c = *a;
+    *a = *b;
+    *b = c;
+}
+
+void reverse(int *v, unsigned int len) {
+    for (unsigned int i = 0; i < (len + 1) / 2; i++) {
+      const int a = i;
+      const int b = len - 1 - i;
+
+      swap(v + a, v + b);
+    }
+}
+
+void createAndFillVector(int*& v, unsigned int len) {
+    v = new int[len];
+    for (unsigned int i = 0; i < len; i++) {
+      v[i] = i;
+    }
+}
+
+int main() {
+    constexpr auto arraySize = 100;
+    int *v = nullptr;
+    // create and reverse the vector of LEN numbers
+    reverse(v, 1000);
+    createAndFillVector(v, arraySize);
```

Comment:
```suggestion
    v = createAndFillVector(arraySize);
```

Reply:
Very good, thanks! It goes in the direction of value semantics, so I will do this transformation also with the other three files.